### PR TITLE
frontend: Log Citra version.

### PR DESCRIPTION
**Pull request by adityaruplaha**
Class: Frontend
Level: Info
Format: "Citra Version: (Nightly/Canary) | (branch)-(commit)"

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3665)

### CODE REVIEW

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -1294,6 +1295,12 @@ void GMainWindow::SetupUIStrings() {
         tr("Citra %1| %2-%3").arg(Common::g_build_name, Common::g_scm_branch, Common::g_scm_desc));
 }
 
+void GMainWindow::LogVersion() {
+    QString version =
+        tr("Citra %1| %2-%3").arg(Common::g_build_name, Common::g_scm_branch, Common::g_scm_desc);
+    NGLOG_INFO(Frontend, ("Citra Version: " + version.toStdString()).c_str());
```
Comment (wwylele) [inline src/citra_qt/main.cpp]: There is no need to use different formatting method from SDL frontend, plus this is not entirely correct: if one names a branch with `{...}` it would be interpreted as formatting placeholder here

Comment (adityaruplaha) [reply src/citra_qt/main.cpp]: Understood. But aren't branch names to be formatted like `branch-name`? _Without braces?_

Comment (wwylele) [reply src/citra_qt/main.cpp]: There is no rule that forbids using of braces in the branch name.

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -1294,6 +1295,12 @@ void GMainWindow::SetupUIStrings() {
         tr("Citra %1| %2-%3").arg(Common::g_build_name, Common::g_scm_branch, Common::g_scm_desc));
 }
 
+void GMainWindow::LogVersion() {
+    QString version =
+        tr("Citra %1| %2-%3").arg(Common::g_build_name, Common::g_scm_branch, Common::g_scm_desc);
```
Comment (baka0815) [inline src/citra_qt/main.cpp]: There is no need to translate this. The name is Citra and shouldn't be translated, the rest are variables which won't get translated.

It's also meaningless to translate this some lines above in ``SetupUIStrings()`` but that's not part of this PR.

File: src/citra_qt/main.cpp (hunk 1 above)
Comment (adityaruplaha) [reply src/citra_qt/main.cpp]: Well, it's just what we do with the window title, don't we? I haven't changed anything. And what do you suggest?

Comment (wwylele) [reply src/citra_qt/main.cpp]: @adityaruplaha Let me clarify my comments. There are **two* issue in your code
 - You don't need, and probably shouldn't, translate a log string. Logging is different from window title, which prefer minimal character set and unchangeable text. Removing the translation meaning that you don't need Qt's formatter at all either
 - Any non-constant string shouldn't pass directly to fmt format string, because of potential `{}` contained in the string. Even if you want to keep the Qt part, you need to write
`NGLOG_INFO(Frontend, "{}", "Citra Version: " + version.toStdString()); // note the "{}"!`

Comment (adityaruplaha) [reply src/citra_qt/main.cpp]: Understood. Will push a new commit fixing these today. Sorry for the delay (will tell ya guys what happened over Discord).

File: src/citra/emu_window/emu_window_sdl2.cpp [hunk 3]
```diff
@@ -167,6 +168,13 @@ void EmuWindow_SDL2::MakeCurrent() {
     SDL_GL_MakeCurrent(render_window, gl_context);
 }
 
+void EmuWindow_SDL2::LogVersion() {
+    NGLOG_INFO(Frontend, "Citra Version: {}",
+               Common::StringFromFormat("Citra %s| %s-%s ", Common::g_build_name,
```
Comment (wwylele) [inline src/citra/emu_window/emu_window_sdl2.cpp]: Could you please just use fmtlib formatting string...

File: src/citra_qt/main.cpp [hunk 4]
```diff
@@ -1294,6 +1295,13 @@ void GMainWindow::SetupUIStrings() {
         tr("Citra %1| %2-%3").arg(Common::g_build_name, Common::g_scm_branch, Common::g_scm_desc));
 }
 
+void GMainWindow::LogVersion() {
+    NGLOG_INFO(Frontend, "Citra Version: {}",
+               Common::StringFromFormat("Citra %s| %s-%s ", Common::g_build_name,
```
Comment (daniellimws) [inline src/citra_qt/main.cpp]: This too like what wwylele said

File: src/citra/emu_window/emu_window_sdl2.cpp [hunk 5]
```diff
@@ -167,6 +168,11 @@ void EmuWindow_SDL2::MakeCurrent() {
     SDL_GL_MakeCurrent(render_window, gl_context);
 }
 
+void EmuWindow_SDL2::LogVersion() {
```
Comment (bunnei) [inline src/citra/emu_window/emu_window_sdl2.cpp]: Why does this need to be it's own function in both of these classes, and not just a single LOG call?

**Comment by adityaruplaha** [comment]
@bunnei Updated.
